src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformRepositoriesHelper.kt

```diff
@@ -149,8 +149,15 @@ class IntelliJPlatformRepositoriesHelper(
         val localPlatformArtifactsPath = providers.localPlatformArtifactsPath(rootProjectDirectory)
         ivyPattern("${localPlatformArtifactsPath.pathString}/[organization]-[module]-[revision].[ext]")
 
-        // As all artifacts defined in Ivy repositories have a full artifact path set as their names, we can use them to locate artifact files
-        artifactPattern("/[artifact]")
+        // "type" may contain an optional absolute path to the actual location of the artifact, in different situations
+        // it may point to a completely unrelated locations on the file system.
+        // "name" is a file name
+        // "ext" is file extension, e.g. ".jar"
+        //
+        // It has to be prefixed by "/" because: If this pattern is not a fully-qualified URL, it will be interpreted
+        // as a file relative to the project directory.
+        val pattern = "[type][artifact]"
+        artifactPattern("/$pattern")
```

Comment:
This is the essence of the fix. Now type contains an optional absolute path. I know that I am somewhat misusing the attribute, but there is nothing else to replace it with. 

The reason why we put tha absolute path into type is that the name should not have it, because artifact name may come up in files like Gradle's verification-metadata.xml 

https://docs.gradle.org/current/userguide/dependency_verification.html

Which will make them not portable between different environments.